torch_geometric_temporal/nn/convolutional/mtgnn.py

```diff
@@ -0,0 +1,445 @@
+            * **idx** (Pytorch Long Tensor) - Input indices.
+            
+        Return types:
+            * **X** (PyTorch Float Tensor) - Output tensor, with shape (batch_size, feature_dim, num_nodes, seq_len).
+        """
+        if self._elementwise_affine:
+            return F.layer_norm(X, tuple(X.shape[1:]), self._weight[:, idx, :], self._bias[:, idx, :], self._eps)
+        else:
+            return F.layer_norm(X, tuple(X.shape[1:]), self._weight, self._bias, self._eps)
+
+    def extra_repr(self):
+        return '{normalized_shape}, eps={eps}, ' \
+            'elementwise_affine={elementwise_affine}'.format(**self.__dict__)
+
+
+class MTGNN(nn.Module):
+    r"""An implementation of the Multivariate Time Series Forecasting Graph Neural Networks.
+    For details see this paper: `"Connecting the Dots: Multivariate Time Series Forecasting with Graph Neural Networks." 
+    <https://arxiv.org/pdf/2005.11650.pdf>`_
+
+    Args:
+        gcn_true (bool) : Whether to add graph convolution layer.
+        buildA_true (bool) : Whether to construct adaptive adjacency matrix.
+        gcn_depth (int) : Graph convolution depth.
+        num_nodes (int) : Number of nodes in the graph.
+        dropout (float, optional) : Droupout rate, default 0.3.
+        subgraph_size (int, optional) : Size of subgraph, default 20.
+        node_dim (int, optional) : Dimension of nodes, default 40.
+        dilation_exponential (int, optional) : Dilation exponential, default 1.
+        conv_channels (int, optional) : Convolution channels, default 32.
+        residual_channels (int, optional) : Residual channels, default 32.
+        skip_channels (int, optional) : Skip channels, default 64.
+        end_channels (int, optional): End channels, default 128.
+        seq_length (int, optional) : Length of input sequence, default 12.
+        in_dim (int, optional) : Input dimension, default 2.
+        out_dim (int, optional) : Output dimension, default 12.
+        layers (int, optional) : Number of layers, default 3.
+        propalpha (float, optional) : Prop alpha, ratio of retaining the root nodes's original states in mix-hop propagation, a value between 0 and 1, default 0.05.
+        tanhalpha (float, optional) : Tanh alpha for generating adjacency matrix, alpha controls the saturation rate, default 3.
+        layer_norm_affline (bool, optional) : Whether to do elementwise affine in Layer Normalization, default True.
+    """
+
+    def __init__(self, gcn_true: bool, buildA_true: bool, gcn_depth: int, num_nodes: int, dropout: float=0.3, subgraph_size: int=20, node_dim: int=40, dilation_exponential: int=1, conv_channels: int=32, residual_channels: int=32, skip_channels: int=64, end_channels: int=128, seq_length: int=12, in_dim: int=2, out_dim: int=12, layers: int=3, propalpha: float=0.05, tanhalpha: float=3, layer_norm_affline: bool=True):
+        super(MTGNN, self).__init__()
+        self._gcn_true = gcn_true
```

Comment:
Separate out the inits of blocks the methods which are smaller and meaningful blocks - 7-8 lines at max